xonsh/tools.py

```diff
@@ -1582,9 +1582,13 @@ def expanduser_abs_path(inp):
 def expand_case_matching(s):
     """Expands a string to a case insensitive globable string."""
     t = []
-    openers = {'[', '{'}
-    closers = {']', '}'}
-    nesting = 0
+    openers = {'[': 'brackets', '{': 'braces'}
+    closers = {']': 'brackets', '}': 'braces'}
+    nesting = {'brackets': 0, 'braces': 0}
+    separators = {'brackets': None, 'braces': ','}
+    brace_prefix = ''
+    brace_args = []
+    cur_brace_arg = ''
```

Comment:
I don't think that it is a good idea to modify this function.  Instead, we probably want a separate function that is independent of the case matching stuff.